app/src/test/java/org/oppia/app/testing/ImageRegionSelectionInteractionViewTest.kt

```diff
@@ -0,0 +1,302 @@
+import org.oppia.app.utility.OnClickableAreaClickedListener
+import org.oppia.app.utility.RegionClickedEvent
+import org.oppia.app.utility.clickPoint
+import org.oppia.data.backends.gae.NetworkModule
+import org.oppia.domain.classify.InteractionsModule
+import org.oppia.domain.classify.rules.continueinteraction.ContinueModule
+import org.oppia.domain.classify.rules.dragAndDropSortInput.DragDropSortInputModule
+import org.oppia.domain.classify.rules.fractioninput.FractionInputModule
+import org.oppia.domain.classify.rules.imageClickInput.ImageClickInputModule
+import org.oppia.domain.classify.rules.itemselectioninput.ItemSelectionInputModule
+import org.oppia.domain.classify.rules.multiplechoiceinput.MultipleChoiceInputModule
+import org.oppia.domain.classify.rules.numberwithunits.NumberWithUnitsRuleModule
+import org.oppia.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.domain.oppialogger.LogStorageModule
+import org.oppia.domain.question.QuestionModule
+import org.oppia.testing.TestAccessibilityModule
+import org.oppia.testing.TestDispatcherModule
+import org.oppia.testing.TestLogReportingModule
+import org.oppia.util.accessibility.FakeAccessibilityManager
+import org.oppia.util.caching.CacheAssetsLocally
+import org.oppia.util.gcsresource.GcsResourceModule
+import org.oppia.util.logging.LoggerModule
+import org.oppia.util.parser.GlideImageLoaderModule
+import org.oppia.util.parser.HtmlParserEntityTypeModule
+import org.oppia.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [StateFragment]. */
+@RunWith(AndroidJUnit4::class)
+@Config(application = ImageRegionSelectionInteractionViewTest.TestApplication::class)
+class ImageRegionSelectionInteractionViewTest {
+
+  @Inject
+  @field:ApplicationContext
+  lateinit var context: Context
+
+  @Inject
+  lateinit var fakeAccessibilityManager: FakeAccessibilityManager
+
+  @Mock
+  lateinit var onClickableAreaClickedListener: OnClickableAreaClickedListener
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    MockitoAnnotations.initMocks(this)
+    FirebaseApp.initializeApp(context)
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testImageRegionSelectionInteractionView_clickRegion3_Region3Clicked() {
+    launch(ImageRegionSelectionTestActivity::class.java).use {
+      it.onActivity { activity ->
+        setUpActivity(activity)
+
+        onView(withId(R.id.clickable_image_view)).perform(
+          clickPoint(0.3f, 0.3f)
+        )
+        onView(allOf(withTagValue(`is`("Region 3"))))
+          .check(
+            matches(isDisplayed())
+          )
+
+        verify(onClickableAreaClickedListener, times(1)).onClickableAreaTouched(
```

Comment:
@BenHenning these doesn't seem to work. Can you tell me  what I might be doing incorrectly